File: src/main/java/com/tpt/api_mbds/model/Match.java
package com.tpt.api_mbds.model;

import com.tpt.api_mbds.repository.MatchRepository;
import org.bson.types.ObjectId;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;
import org.springframework.data.mongodb.core.mapping.DBRef;
import java.util.Date;
import java.util.List;
import java.util.Random;
import java.util.concurrent.ThreadLocalRandom;

@Document(collection="match")
public class Match {
    @Autowired
    MatchRepository matchRepository;

    @Id
    private String id;

    @DBRef
    private  Equipe equipe1;
    @DBRef
    private  Equipe equipe2;

    //private List<Equipe> equipes;

    private Date date;
    private String lieu;
    private String etat;
    private Integer scoreEquipe1;
    private Integer scoreEquipe2;
    private Integer cornerEquipe1;
    private Integer cornerEquipe2;
    private Integer possessionEquipe1;
    private Integer possessionEquipe2;


    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public Equipe getEquipe1() {
        return equipe1;
    }

    public void setEquipe1(Equipe equipe1) {
        this.equipe1 = equipe1;
    }

    public Equipe getEquipe2() {
        return equipe2;
    }

    public void setEquipe2(Equipe equipe2) {
        this.equipe2 = equipe2;
    }

    public Date getDate() {
        return date;
    }

    public void setDate(Date date) {
        this.date = date;
    }

    public String getLieu() {
        return lieu;
    }

    public void setLieu(String lieu) {
        this.lieu = lieu;
    }

    public String getEtat() {
        return etat;
    }

    public void setEtat(String etat) {
        this.etat = etat;
    }

    public Integer getScoreEquipe1() {
        return scoreEquipe1;
    }

    public void setScoreEquipe1(Integer scoreEquipe1) {
        this.scoreEquipe1 = scoreEquipe1;
    }

    public Integer getScoreEquipe2() {
        return scoreEquipe2;
    }

    public void setScoreEquipe2(Integer scoreEquipe2) {
        this.scoreEquipe2 = scoreEquipe2;
    }

    public Integer getCornerEquipe1() {
        return cornerEquipe1;
    }

    public void setCornerEquipe1(Integer cornerEquipe1) {
        this.cornerEquipe1 = cornerEquipe1;
    }

    public Integer getCornerEquipe2() {
        return cornerEquipe2;
    }

    public void setCornerEquipe2(Integer cornerEquipe2) {
        this.cornerEquipe2 = cornerEquipe2;
    }

    public Integer getPossessionEquipe1() {
        return possessionEquipe1;
    }

    public void setPossessionEquipe1(Integer possessionEquipe1) {
        this.possessionEquipe1 = possessionEquipe1;
    }

    public Integer getPossessionEquipe2() {
        return possessionEquipe2;
    }

    public void setPossessionEquipe2(Integer possessionEquipe2) {
        this.possessionEquipe2 = possessionEquipe2;
    }

    public Match() {

    }

   



    public Match(String id, Equipe equipe1, Equipe equipe2, Date date, String lieu, String etat, Integer scoreEquipe1, Integer scoreEquipe2, Integer cornerEquipe1, Integer cornerEquipe2, Integer possessionEquipe1, Integer possessionEquipe2) {
        this.id = id;
        this.equipe1 = equipe1;
        this.equipe2 = equipe2;
        this.date = date;
        this.lieu = lieu;
        this.etat = etat;
        this.scoreEquipe1 = scoreEquipe1;
        this.scoreEquipe2 = scoreEquipe2;
        this.cornerEquipe1 = cornerEquipe1;
        this.cornerEquipe2 = cornerEquipe2;
        this.possessionEquipe1 = possessionEquipe1;
        this.possessionEquipe2 = possessionEquipe2;
    }

    public Match(Equipe equipe1, Equipe equipe2, Date date, String lieu, String etat, Integer scoreEquipe1, Integer scoreEquipe2, Integer cornerEquipe1, Integer cornerEquipe2, Integer possessionEquipe1, Integer possessionEquipe2) {
        this.equipe1 = equipe1;
        this.equipe2 = equipe2;
        this.date = date;
        this.lieu = lieu;
        this.etat = etat;
        this.scoreEquipe1 = scoreEquipe1;
        this.scoreEquipe2 = scoreEquipe2;
        this.cornerEquipe1 = cornerEquipe1;
        this.cornerEquipe2 = cornerEquipe2;
        this.possessionEquipe1 = possessionEquipe1;
        this.possessionEquipe2 = possessionEquipe2;
    }

    public void startmatch() {
        this.setEtat("2");
    }

    public void endmatch(Match m,MatchRepository repository) {
        m.setEtat("3");
        m.setscores();
        this.matchRepository=repository;

        Match _match = m;
        //System.out.println("ITO ILAY MATCH ID "+_match.getId());
        _match.setEquipe1(m.getEquipe1());
        //System.out.println("ITO ILAY MATCH EQUIPE1 "+_match.getEquipe1().getId());
        _match.setEquipe2(m.getEquipe2());
        _match.setDate(m.getDate());
        _match.setEtat(m.getEtat());
        // System.out.println("ITO ILAY MATCH ETAT "+_match.getEtat());
        _match.setLieu(m.getLieu());
        _match.setScoreEquipe1(m.getScoreEquipe1());
        _match.setScoreEquipe2(m.getScoreEquipe2());
        _match.setCornerEquipe1(m.getCornerEquipe1());
        _match.setCornerEquipe2(m.getCornerEquipe2());
        _match.setPossessionEquipe1(m.getPossessionEquipe1());
        _match.setPossessionEquipe2(m.getPossessionEquipe2());
        matchRepository.save(_match);


    }

    public void setscores() {
        Random random = new Random();
        int randomNum = ThreadLocalRandom.current().nextInt(30, 100 + 1);
        this.setScoreEquipe1( random.nextInt(8));
        this.setScoreEquipe2( random.nextInt(8));
        this.setCornerEquipe1( random.nextInt(8));
        this.setCornerEquipe2( random.nextInt(8));
        this.setPossessionEquipe1( random.nextInt(randomNum));
        this.setPossessionEquipe2( 100-this.getPossessionEquipe1());

    }

}

File: winform/projetParis/projetParis/Model/Match.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;
using System;
using System.Collections.Generic;
using System.Text;

namespace projetParis.Model
{
    class Match
    {
        [MongoDB.Bson.Serialization.Attributes.BsonId]
        [BsonRepresentation(BsonType.ObjectId)]
        private string id;
        private Equipe equipe1;
        private Equipe equipe2;
        private DateTime date;
        private string lieu;
        private string etat;
        private int scoreEquipe1;
        private int scoreEquipe2;
        private int cornerEquipe1;
        private int cornerEquipe2;
        private int possessionEquipe1;
        private int possessionEquipe2;

        public string Id { get => id; set => id = value; }
        public DateTime Date { get => date; set => date = value; }
        public string Lieu { get => lieu; set => lieu = value; }
        public string Etat { get => etat; set => etat = value; }
        public int ScoreEquipe1 { get => scoreEquipe1; set => scoreEquipe1 = value; }
        public int ScoreEquipe2 { get => scoreEquipe2; set => scoreEquipe2 = value; }
        public int CornerEquipe1 { get => cornerEquipe1; set => cornerEquipe1 = value; }
        public int CornerEquipe2 { get => cornerEquipe2; set => cornerEquipe2 = value; }
        public int PossessionEquipe1 { get => possessionEquipe1; set => possessionEquipe1 = value; }
        public int PossessionEquipe2 { get => possessionEquipe2; set => possessionEquipe2 = value; }
        internal Equipe Equipe1 { get => equipe1; set => equipe1 = value; }
        internal Equipe Equipe2 { get => equipe2; set => equipe2 = value; }

        public Match()
        {
        }

        public Match(string id, DateTime date, string lieu, string etat, int scoreEquipe1, int scoreEquipe2, int cornerEquipe1, int cornerEquipe2, int possessionEquipe1, int possessionEquipe2, Equipe equipe1, Equipe equipe2)
        {
            Id = id;
            Date = date;
            Lieu = lieu;
            Etat = etat;
            ScoreEquipe1 = scoreEquipe1;
            ScoreEquipe2 = scoreEquipe2;
            CornerEquipe1 = cornerEquipe1;
            CornerEquipe2 = cornerEquipe2;
            PossessionEquipe1 = possessionEquipe1;
            PossessionEquipe2 = possessionEquipe2;
            Equipe1 = equipe1;
            Equipe2 = equipe2;
        }
    }
}
